src/services/groups/index.ts

```diff
@@ -66,13 +67,31 @@ export const fetchGroups = async () => {
 export const getGroupsPage = async (
   count: number,
   startIndex: number = 1,
+  me: boolean = false,
   filter?: string
 ) => {
-  let url = `${BASE_URL}/iam/group/search?count=${count}&startIndex=${startIndex}`;
-  if (filter) {
-    url += `&filter=${filter}`;
+  if (me) {
+    // modo osceno per recuperare i gruppi dell'utente
+    const meObj = await fetchMe();
+    let groups: Paginated<Group> = {
+      totalResults: 0,
+      itemsPerPage: count,
+      startIndex: startIndex,
+      Resources: []
+    };
+    for(const g of meObj?.groups ?? []) {
+      groups.Resources.push(await fetchGroup(g.value));
```

Comment:
Here we have a blocking for loop where each iteration wait for the previous one to be finished.
Replace with `Promise.all([list_of_promises])`